server/lib/puppet-languageserver/puppet_parser_helper.rb

```diff
@@ -1,15 +1,28 @@
 module PuppetLanguageServer
   module PuppetParserHelper
-    def self.remove_char_at(content, line_offsets, line_num, char_num)
+    def self.remove_chars_starting_at(content, line_offsets, line_num, char_num, move_offset)
       line_offset = line_offsets[line_num]
       raise if line_offset.nil?
 
       # Remove the offending character
-      new_content = content.slice(0, line_offset + char_num - 1) + content.slice(line_offset + char_num, content.length - 1)
+      new_content = content.slice(0, line_offset + char_num + move_offset) + content.slice(line_offset + char_num, content.length - 1)
 
       new_content
     end
 
+    def self.remove_char_at(content, line_offsets, line_num, char_num)
+      remove_chars_starting_at(content, line_offsets, line_num, char_num, -1)
+    end
+
+    def self.get_char_at(content, line_offsets, line_num, char_num)
+      line_offset = line_offsets[line_num]
+      raise if line_offset.nil?
+
+      absolute_offset = line_offset + (char_num - 1)
```

Comment:
When typing the "n" in this manifest, I ran into the following problem.
```
user { 'Bob' :
  ensure => 'present',
  n
}
```

Subtracting 1 got me the character I wanted, but maybe I should have gone back one character in the `remove_word` method instead?
```
    18: def self.get_char_at(content, line_offsets, line_num, char_num)
    19:   line_offset = line_offsets[line_num]
    20:   raise if line_offset.nil?
    21:
    22:   binding.pry
    23:   absolute_offset = line_offset + (char_num - 1)
    24:
 => 25:   content[absolute_offset]
    26: end

[1] pry(PuppetLanguageServer::PuppetParserHelper)> content[absolute_offset]
=> "n"
[2] pry(PuppetLanguageServer::PuppetParserHelper)> content[absolute_offset + 1]
=> "\n"
```